libraries/AP_HAL_SITL/Semaphores.cpp

```diff
@@ -24,7 +24,10 @@ bool Semaphore::give()
     if (pthread_mutex_unlock(&_lock) != 0) {
         AP_HAL::panic("Bad semaphore usage");
     }
-    owner = (pthread_t)-1;
+    take_count--;
```

Comment:
i think the change to the take_count needs to come before the pthread_mutex_unlock(), otherwise it races with another user of the semaphore

Reply:
Fixed.